### PR TITLE
feat: Added Morse Code

### PR DESCRIPTION
**Pull request by imdeep2905**
#### Description of Change

Added Morse Code as ```ciphers/morse_code.cpp```.

#### Checklist

- [X] Added description of change
- [X] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [X] Added tests and example, test must pass
- [X] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [X] Relevant documentation/comments is changed or added
- [X] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [X] Search previous suggestions before making a new one, as yours may be a duplicate.
- [X] I acknowledge that all my contributions will be made under the project's license.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1322"></a>

### CODE REVIEW

File: ciphers/morse_code.cpp [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+    return decrypted_text;  // Returning decrypted text
+}
+}  // namespace morse
+}  // namespace ciphers
+
+/**
+ * Function to test above algorithm
+ */
+void test() {
```
Comment (Panquesito7) [inline ciphers/morse_code.cpp:244]:
```suggestion
static void test() {
```

File: ciphers/morse_code.cpp [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+        decrypted_text += ciphers::morse::morse_to_char(s);
+    }
+
+    return decrypted_text;  // Returning decrypted text
+}
+}  // namespace morse
+}  // namespace ciphers
+
+/**
+ * Function to test above algorithm
```
Comment (Panquesito7) [inline ciphers/morse_code.cpp:242]:
```suggestion
 * @brief Function to test above algorithm
 * @returns void
```

File: ciphers/morse_code.cpp [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    std::string text2 = "abcdefghijklmnopqrstuvwxyz";
+    std::string encrypted2 = ciphers::morse::encrypt(text2);
+    std::string decrypted2 = ciphers::morse::decrypt(encrypted2);
+    assert(text2 == decrypted2);
+    std::cout << "Original text : " << text2 << std::endl;
+    std::cout << "Encrypted text : " << encrypted2 << std::endl;
+    std::cout << "Decrypted text : " << decrypted2 << std::endl;
+}
+
+/** Driver Code */
```
Comment (Panquesito7) [inline ciphers/morse_code.cpp:263]:
```suggestion
/**
 * @brief Main function
 * @returns 0 on exit
 */
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
👍 Well written and documented code.

File: ciphers/morse_code.cpp [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+namespace ciphers {
+/** \namespace morse
+ * \brief Functions for [Morse Code]
+ * (https://en.wikipedia.org/wiki/Morse_code).
+ */
+namespace morse {
+/**
+ * Get the morse representation for given character.
+ * @param c Character
+ * @return morse representation string of character
```
Comment (Panquesito7) [inline ciphers/morse_code.cpp:31]:
```suggestion
 * @returns morse representation string of character
```

File: ciphers/morse_code.cpp [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+            return "-----";
+        default:
+            std::cerr << "Found invalid character: " << c << ' ' << std::endl;
+            std::exit(0);
+    }
+}
+/**
+ * Get character from the morse representation.
+ * @param s Morse representation
+ * @return corresponding character
```
Comment (Panquesito7) [inline ciphers/morse_code.cpp:116]:
```suggestion
 * @returns corresponding character
```

File: ciphers/morse_code.cpp [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+        return '0';
+    } else {
+        std::cerr << "Found invalid Morse code: " << s << ' ' << std::endl;
+        std::exit(0);
+    }
+}
+/**
+ * Encrypt given text using morse code.
+ * @param text text to be encrypted
+ * @return new encrypted text
```
Comment (Panquesito7) [inline ciphers/morse_code.cpp:200]:
```suggestion
 * @returns new encrypted text
```

File: ciphers/morse_code.cpp [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+    // to morse representation
+    for (const char &c : text) {
+        encrypted_text += ciphers::morse::char_to_morse(c) + " ";
+    }
+    return encrypted_text;  // Returning encrypted text
+}
+/**
+ * Decrypt given morse coded text.
+ * @param text text to be decrypted
+ * @return new decrypted text
```
Comment (Panquesito7) [inline ciphers/morse_code.cpp:214]:
```suggestion
 * @returns new decrypted text
```

**Review by Panquesito7 — APPROVED** [review]
Amazing work, I like how you've made the code. 😄 👍
Code and documentation is pretty good and refined; LGTM. 🙂

**Comment by imdeep2905** [comment]
Hey, Any idea why LGTM failed ?
Shows error:
```This pull request can't be analyzed because the requested commit is no longer present in the repository```

**Comment by Panquesito7** [comment]
> Hey, Any idea why LGTM failed ?
> Shows error:
> `This pull request can't be analyzed because the requested commit is no longer present in the repository`

When GitHub Actions force-pushed, LGTM couldn't check
the commits (as after every force-push, sometimes commits disappear).

I re-ran the checks, it shall work properly now. 🙂

**Review by kvedala — APPROVED** [review]
👍  Well done

**Pull request merged**